### PR TITLE
Map INTERVAL types to Python types

### PR DESCRIPTION
**Pull request by hovaesco**
## Description
Resolves https://github.com/trinodb/trino-python-client/issues/356
- Map INTERVAL YEAR TO MONTH to `relativedelta`
- Map INTERVAL DAY TO SECOND to `timedelta`

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: tests/integration/test_types_integration.py [hunk 1]
```diff
@@ -737,13 +739,30 @@ def create_timezone(timezone_str: str) -> tzinfo:
         return ZoneInfo(timezone_str)
 
 
-def test_interval(trino_connection):
+def test_interval_year_to_month(trino_connection):
     SqlTest(trino_connection) \
         .add_field(sql="CAST(null AS INTERVAL YEAR TO MONTH)", python=None) \
+        .add_field(sql="INTERVAL '10' YEAR", python=relativedelta(years=10)) \
```
Comment (hashhar) [inline tests/integration/test_types_integration.py:745]: Please test for min/max valid trino value and min/max valid python values too.

Comment (hashhar) [reply tests/integration/test_types_integration.py:745]: Does this test verify both directions? i.e. converting from Trino string representation to Python object AND converting from Python object to Trino SQL (e.g. queries with parameters?)?

**Review by hashhar — COMMENTED** [review]
LGTM % questions

File: tests/integration/test_types_integration.py [hunk 2]
```diff
@@ -737,13 +739,30 @@ def create_timezone(timezone_str: str) -> tzinfo:
...
-        .add_field(sql="INTERVAL '-2' DAY", python='-2 00:00:00.000') \
+        .add_field(sql="INTERVAL '2' DAY", python=timedelta(days=2)) \
+        .add_field(sql="INTERVAL '-2' DAY", python=timedelta(days=-2)) \
+        .add_field(sql="INTERVAL '-2' SECOND", python=timedelta(seconds=-2)) \
+        .add_field(sql="INTERVAL '1 11:11:11.116555' DAY TO SECOND",
+                   python=timedelta(days=1, seconds=40271, microseconds=116000)) \
+        .add_field(sql="INTERVAL '-5 23:59:57.000' DAY TO SECOND", python=timedelta(days=-6, seconds=3)) \
+        .add_field(sql="INTERVAL '12 10:45' DAY TO MINUTE", python=timedelta(days=12, seconds=38700)) \
+        .add_field(sql="INTERVAL '45:32.123' MINUTE TO SECOND", python=timedelta(seconds=2732, microseconds=123000)) \
+        .add_field(sql="INTERVAL '32.123' SECOND", python=timedelta(seconds=32, microseconds=123000)) \
```
[the remark this answers is not in the document]
Comment (hashhar) [reply tests/integration/test_types_integration.py:765]: As discussed offline let's check what the JDBC driver does when a Trino value is too large/small to fit into corresponding Java type. If it throws an error then the existing implementation we have here seems fine to me.

[the remark this answers is not in the document]
Comment (hashhar) [reply tests/integration/test_types_integration.py:765]: We do use `TrinoDataError` in some places where values from Trino cannot be mapped to Python.

[the remark this answers is not in the document]
Comment (damian3031) [reply tests/integration/test_types_integration.py]: Added test cases for edge scenarios. Raised TrinoDataError when Trino value is too large/small. PTAL @hashhar

**Review by hashhar — APPROVED** [review]

**Pull request merged**